L1Trigger/TrackTrigger/src/TTStubAlgorithm_official.cc

```diff
@@ -124,7 +126,100 @@ void TTStubAlgorithm_official< Ref_Phase2TrackerDigi_ >::PatternHitCorrelation(
   {
     aConfirmation = true;
     aDisplacement = dispI; /// In HALF-STRIP units!
-    anOffset = offsetI; /// In HALF-STRIP units!
+    anOffset      = offsetI; /// In HALF-STRIP units!
+    anROffset     = static_cast<float>(offsetD); /// In HALF-STRIP units!
+    anHardBend    = this->degradeBend(isPS, window, (aDisplacement - anOffset)); // In strips units
   } /// End of stub is accepted
 
 }
+
+
+//--- Does the actual work of degrading the bend. (based on I.Tomalin's code)
+
+template< >   
+float TTStubAlgorithm_official< Ref_Phase2TrackerDigi_ >::degradeBend(bool psModule, int window, int bend) const {
+ 
+
+  // Number of bits used to encoded bend output by FE electronics.
+  const unsigned int bitsPS_ = 3;
+  const unsigned int bits2S_ = 4;
+
+  // Number of degraded bend values should correspond to 3 bits (PS modules) or 4 bits (2S modules),
+  // so measuring everything in half-strip units, max integer "window" size that can be encoded without
+  // compression given by 2*window+1 <= pow(2,B), where B is number of bits.
+  // Hence no compression required if window cut is abs(b) <= 3 (PS) or 7 (2S). Must introduce one merge for
+  // each 1 unit increase in "window" beyond this.
+
+  // Bend is measured with granularity of 0.5 strips.
+  // Convert it to integer measured in half-strip units for this calculation!
+
+  float degradedB;
+  unsigned int numBends = 2*window + 1;
+  unsigned int numAllowed = (psModule)  ?  pow(2, bitsPS_)  :  pow(2, bits2S_);
+
+  // Existance of bend = 0 means can only use an odd number of groups.
+  numAllowed -= 1; // 7 or 15
```

Comment:
Confused here... in what way is -1 equivalent to 7 or 15?

Reply:
Clarified by additional comment